api/filters.go

```diff
@@ -153,8 +153,8 @@ func (api *API) submitFilter(w http.ResponseWriter, r *http.Request) {
 	// Just chosen a reasonable looking one.
 	// TODO: RAISE STATE in a round table.
 	filterOutput := &model.FilterOutput{
-		ID:    filter.Dataset.ID,
-		State: "submitted",
+		FilterID: filter.Dataset.ID,
+		State:    "submitted",
```

Comment:
You're using a const for completed state, you should probably be consistent and use for others too